.github/workflows/build.yml

```diff
@@ -0,0 +1,99 @@
+name: IntuneAppBuilder Deployment
+
+# TODO: Remove these comments once reviewed
+# on:
+#   schedules:
+#   - cron: "30 7 * * *"
+#     displayName: Daily job
+#     branches:
+#       include:
+#       - master
+#     always: true
+#   push:
+#     branches:
+#       - master
+#   pull_request:
+#     types: [opened, synchronize, reopened, closed]
+#     branches:
+#       - master
+
+on: workflow_dispatch
+
+env:
+  BuildConfiguration: 'Release'
+  Verbosity: 'normal'
+  ArtifactFile: 'output.nupkg'
+
+defaults:
+  runs-on: windows-latest
+  run:
+    shell: pwsh
+
+jobs:
+  Build:
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.301
+
+      - name: dotnet restore
+        run: dotnet restore --configuration $(Env:BuildConfiguration) --verbosity $(Env:Verbosity)
+      
+      - name: dotnet build
+        run: dotnet build --configuration $(Env:BuildConfiguration) --verbosity $(Env:Verbosity)
+
+      - name: dotnet pack
+        run: dotnet pack --configuration $(Env:BuildConfiguration) --verbosity $(Env:Verbosity) --no-build *.sln
```

Comment:
I don't think you need *.sln in here
Also, I think instead of using the publish step at all, we can do everything we need here:
`dotnet pack ...... --no-build -o ./artifacts`